Mathlib/AlgebraicTopology/SimplicialObject/Coskeletal.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2024 Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Emily Riehl, Joël Riou
+-/
+import Mathlib.AlgebraicTopology.SimplicialObject.Basic
+import Mathlib.CategoryTheory.Functor.KanExtension.Adjunction
+import Mathlib.CategoryTheory.Functor.KanExtension.Basic
+
+/-!
+# Coskeletal simplicial objects
+
+The identity natural transformation exhibits a simplicial object `X` as a right extension of its
+restriction along `(Truncated.inclusion (n := n)).op` recorded by `rightExtensionInclusion X n`.
+
+The simplicial object `X` is *n-coskeletal* if `(rightExtensionInclusion X n)` is a right Kan
+extension.
+
+When the ambient category admits right Kan extensions along `(Truncated.inclusion (n := n)).op`,
+then when `X` is `n`-coskeletal, the unit of `coskAdj n` defines an isomorphism:
+`isoCoskOfIsCoskeletal : X ≅ (Truncated.cosk n).obj ((truncation n).obj X)`.
+
+TODO: Prove that `X` is `n`-coskeletal whenever a certain canonical cone is a limit cone.
+-/
+
+open Opposite
+
+open CategoryTheory
+
+open CategoryTheory.Limits CategoryTheory.Functor SimplexCategory
+
+universe v u v' u'
+
+namespace CategoryTheory
+
+namespace SimplicialObject
+variable {C : Type u} [Category.{v} C]
+variable (X : SimplicialObject C) (n : ℕ)
+
+namespace Truncated
+
+/-- The identity natural transformation exhibits a simplicial set as a right extension of its
+restriction along `(Truncated.inclusion (n := n)).op`.-/
+@[simps!]
+def rightExtensionInclusion :
+    RightExtension (Truncated.inclusion (n := n)).op
```

Comment:
Given how often you need to use a named argument to specify the `n` argument of `Truncated.inclusion`, perhaps it should be made explicit.

Reply:
Done.

Reply:
Done.